packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -27,6 +27,10 @@ import 'message_codecs.dart';
 /// The logical identity of the channel is given by its name. Identically named
 /// channels will interfere with each other's communication.
 ///
+/// Applications can typically assume messages are delivered in the same order
+/// as they're sent, unless a custom [BinaryMessenger] that does not guarantee
+/// FIFO ordering is used.
```

Comment:
"can typically" and "unless a custom [BinaryMessenger] that does not guarantee FIFO ordering" makes the whole thing an empty promise. So, can I or can I not rely on the order? What is this text trying to tell me? I think we should either strongly guarantee the FIFO order (make it clear that it's safe to rely on it, and treat any deviation from it as a serious bug) or ask our developers to never rely on it.

To that end I would also update the dartdoc on `BinaryMessenger.send` to require FIFO delivery (or explicitly allow the implementation to choose the order).

To avoid flaky behavior I'd prefer we strongly guarantee FIFO, unless we have good use-cases not to.

Reply:
Yea, this rubbed me the wrong way except we can't technically make the guarantee.  I do think this is an improvement over nothing.  I like the idea or adding FIFO note to BinaryMessenger.  But again we can't enforce it is FIFO.  Maybe we could say that all of Flutter's BinaryMessengers are FIFO so people understand that they'd really not get into this position unless they really go out of their way?

Reply:
Yeah in bindings.dart I worded it stronger for `createBinaryMessenger` "When overriding this method, be sure the [BinaryMessenger]
  /// implementation guarantees FIFO delivery." since that's the messenger all the builtin channels use. 

I can change the paragraph to say that all our builtin channels having fifo ordering.